### DISCUSSION TITLE
빌드 경량화

### DISCUSSION THREAD

**Discussion by TopSonicK** [hamonikr/nimf#90] [Ideas]
현재 nimf의 빌드는 기본으로 anthy, m17n, rime을 모두 포함하여 빌드하게 기본 구성이 돼 있고 사용자가 불필요할 경우 --disable-anthy, --disable-m17n, --diable-rime 을 이용하여 비활성화 하게 되어 있습니다.

이런 빌드 상황은 기본적으로 필요한 종속성도 많아지게 되고, 불필요한 자원을 더 많이 사용하게 되는 점이 있는데다가 빌드가 꼬일 가능성을 높일 수 있다고 생각합니다.

혹시 빌드 방식을
nimf 엔진(기본빌드) - 언어(한글/anthy/m17n/rime)을 플러그인 형태(nimf-hangul, nimf-anthy, nimf-rime)로 빌드하는 방식은 어떻게 생각하실까요?

m17n도 통으로 m17n으로 빌드할 것이 아니라 17개를 각각의 플러그인으로 쪼개서(ex nimf-m17n-ar nimf-m17nj-ath 등) 사용자가 필요한 언어를 선택하는 방식이 조금 더 사용자입장에서는 자원 효율적인 방법이라고 생각하는데 어떻게 생각하시나요?

**Comment by chaeya** [comment]
데비안 패키지는 이미 한국어, 중국어, 일본어, 다른언어로 패키지를 분리해서 제공하고 있습니다.

이 의견은 소스코드를 컴파일할때 옵션으로 다음과 같이 사용하는것을 의미하나요?

```
# 전체 언어 빌드
./configure --enable-plugins=all

# 언어를 선택해서 빌드
./configure --enable-plugins=hangul,anthy

# 코어만 빌드
./configure --enable-plugins=none
```
